django_socio_grpc/protobuf/generation_plugin.py

```diff
@@ -0,0 +1,284 @@
+            not hasattr(service, "pagination_class")
+            or not service.pagination_class
+            and not grpc_settings.DEFAULT_PAGINATION_CLASS
+        ):
+            logger.debug(
+                "You are using PaginationGenerationPlugin but no pagination_class as been found on the service. Please set a pagination_class in the service or set it globally with DEFAULT_PAGINATION_CLASS setting."
+            )
+            return False
+
+        if (
+            grpc_settings.PAGINATION_BEHAVIOR
+            == FilterAndPaginationBehaviorOptions.METADATA_STRICT
+        ):
+            logger.debug(
+                "You are using PaginationGenerationPlugin but PAGINATION_BEHAVIOR settings is set to 'METADATA_STRICT' so it will have no effect. Please change PAGINATION_BEHAVIOR."
+            )
+            return False
+
+        return True
+
+
+@dataclass
+class AsListGenerationPlugin(BaseGenerationPlugin):
+    """
+    Plugin allowing to encapsulate a message into an other message and put it as a repeated field to return multiple instance.
+    Do not use directly, use one of it's subclass RequestAsListGenerationPlugin, ResponseAsListGenerationPlugin or RequestAndResponseAsListGenerationPlugin
+    """
+
+    list_field_name: str = "results"
+
+    def transform_message_to_list(
+        self, service: Type["Service"], proto_message: ProtoMessage, list_name: str
+    ) -> ProtoMessage:
+        try:
+            list_field_name = proto_message.serializer.Meta.message_list_attr
+        except AttributeError:
+            list_field_name = self.list_field_name
+
+        fields = [
+            ProtoField(
+                name=list_field_name,
+                field_type=proto_message,
+                cardinality=FieldCardinality.REPEATED,
+            ),
+        ]
+
+        if hasattr(service, "pagination_class"):
+            fields.append(
+                ProtoField(
+                    name="count",
+                    field_type="int32",
+                )
+            ),
+
+        if grpc_settings.SEPARATE_READ_WRITE_MODEL:
+            pass
+
+        list_message = ProtoMessage(
+            name=list_name,
+            fields=fields,
+        )
+
+        # INFO - AM - If the original proto message is a serializer then we keep the comment at the serializer level. Else we put them at the list level
+        if not proto_message.serializer:
+            list_message.comments = proto_message.comments
+            proto_message.comments = None
+
+        return list_message
+
+
+class RequestAsListGenerationPlugin(AsListGenerationPlugin):
+    """
+    Transform a request ProtoMessage in a list ProtoMessage
+    """
+
+    def transform_request_message(
+        self,
+        service: Type["Service"],
+        proto_message: ProtoMessage,
+        message_name_constructor: MessageNameConstructor,
+    ) -> ProtoMessage:
+        request_constructed_name = message_name_constructor.request_constructed_name
```

Comment:
the message constructor should be the one responsible for this with a .construct_list_name() for example